### ISSUE TITLE
Dropdown: accessibility improvements

### ISSUE THREAD

**Issue by scriby** [OfficeDev/office-ui-fabric-react#1452]
I recorded a video of using the dropdown component on Windows 10 with Narrator + Edge: 

https://www.screencast.com/t/qbAgjdA1zI

Once the dropdown is selected, navigating with the next/previous item (capslock + left arrow, capslock + right arrow) takes you to a few elements within the dropdown:

* A span tag, which contains
* A span tag which has the option text
* The caret icon

I think we should simply mark the caret icon with aria-hidden=true and role=presentation, because they can already open the dropdown by using the "main" component, so we don't need this icon to be the accessibility tree.

There may not be a good solution for preventing it from stopping on the two span tags. Might be possible to combine the two spans into a single tag?

**Comment by c-w** [comment]
I made a pull request to stop screen readers from announcing the caret icon.

Skipping the announcing of the inner span is a bigger challenge. The dropdown currently looks like a bit like this:

```html
<Dropdown aria-describedby="container">
  <span id="container">
    <DropdownEntry id="entry" />
  </span>
</Dropdown>
```

We could skip the inner span if we just set the aria-describedby to "entry". However, the DropdownEntry can be created by a custom rendering function passed into the Dropdown component over which we don't have control in which case we have no easy way of getting an id for the DropdownEntry.

Here's a potential way out of this situation: we add an optional callback `(dropDownContent: HTMLElement) => void` to the DropdownEntry rendering function and document that this callback needs to be called when the DropdownEntry mounts (e.g. via `ref`). The callback can then extract the DropdownEntry's id (or generate a new one if necessary) and set the Dropdown's aria-describedby. This would be a backwards-compatible API change. It will improve the accessibility experience for everyone who relies on the default DropdownEntry rendering and keep the current broken behavior for all clients who use custom rendering.

This is not a very elegant solution so hopefully someone with more React experience can come up with a better way of fixing this.

**Comment by scriby** [comment]
Ultimately I think it's ok to not fix the double-span issue. Maybe the technology will catch up at some point... for instance, VoiceOver doesn't announce those extra elements.

**Comment by c-w** [comment]
Given that @dzearing merged the aria-attribute improvements: can we treat the double-span issue as a wont-fix and close this?
